docs/_tutorials/read-only-json-api.md

```diff
@@ -0,0 +1,321 @@
+* `/categories/{category}` --- an index of posts under a category
+
+Let's dive in!
+
+---
+
+# Setup
+
+First, you need to go through an [installation]({% link _docs/installation.md %}) process. Having that, you can initialize a new Jekyll project. In your directory of choice, invoke the following:
+
+```sh
+bundle init
+bundle config set --local path 'vendor/bundle'
+bundle add jekyll
+bundle exec jekyll new --force --skip-bundle .
+bundle install
+```
+
+* Initialize a new Ruby [bundle](https://bundler.io).
+* Instruct Bundler to install dependencies locally.
+* Add Jekyll as a dependency.
+* Create a new scaffold for the project. 
+* Install dependencies.
+
+Now, you need to do some clean up. 
+
+---
+
+# Configuration
+
+There are some things in the bundle that you won't need:
+
+```
+.
+├── site
+│   ├── index.markdown
+│   ├── about.markdown
+│   ├── 404.html
+│   ├── _posts
+│   │   └── 2020-12-19-welcome-to-jekyll.markdown
+```
+
+Unless you self-host your site, **you can rely on a generic 404 response from your server**. You no longer need `index` --- you will generate a custom one later. Generating pages like `about` is out of scope of this tutorial, but in case you need them, they behave the same way that posts do.
+
+Under `_posts`, create two new documents. Remember that Jekyll requires you to **prepend filenames with a date**:
+
+`2020–12–20-the-first-post.md`
+
+```md
+---
+title:  "The First Post"
+category: update
+custom-property: "My custom property"
+---
+1
+```
+
+Please note that you include a `custom-property` that you will fetch later on. 
+
+`2020–12–20-the-second-post.md`
+
+```md
+---
+title:  "The Second Post"
+categories: ["update", "tutorial"]
+---
+2
+```
+
+This one belongs to two categories --- `update` and `tutorial`, and doesn't include the custom property. Because the categories overlap, it will be possible to test getting a list of posts from a category.
+
+You'll also need a couple of layouts. Just add these empty files for now --- you'll fill (one of) them later:
+
+```
+.
+├── site
+│   ├── _layouts
+│   │   ├── category.html
+│   │   └── default.html
+```
+
+The last thing is to link the `default` layout in `config.yml`:
+
+```yml
+defaults:
+  -
+    scope:
+      path: ""
+    values:
+      layout: "default"
+```
+
+Since you won't have any pages in your project, leave the `path` to `""` in order to [cover]({% link _docs/configuration/front-matter-defaults.md %}) every file.
+
+---
+
+# Convert Posts to JSON
+My approach is a little hacky --- it requires you to specify a JSON output format in a HTML template. The only thing you miss is that your text editor will most probably not recognize the markup file as a properly formatted JSON document. 
+
+## Define the Template
+Before you do it, it's worth noting that you can list all the available properties with this little template, which returns an array of available keys:
+
+```liquid
+{% raw %}{{ page.keys | jsonify }}{% endraw %}
```

Comment:
I don't know how other maintainers feel, but I find the code much easier to read and maintain by wrapping the entire file in `raw`.

```
---
---
{% raw %}

## Heading

content

...

Bottom of page

{% endraw %}
```

Reply:
Fixed. I agree, it's nicer, although I get a warning when I do it:

```
Excerpt modified in _tutorials/read-only-json-api.md!
Found a Liquid block containing the excerpt separator "\n\n". 
The block has been modified with the appropriate closing tag.
Feel free to define a custom excerpt or excerpt_separator in the
document's Front Matter if the generated excerpt is unsatisfactory.
```
Any idea how to resolve it? It's just a warning, but I feel that it should be addressed anyways.

Reply:
Hmm I have used raw tag frequently including for an entire page and have not seen that warning before.

An excerpt is usually in the context of a post but I guess applies to this tutorial too. The excerpt is the first paragraph and it will have a blank line after it which is the \n\n

After rewriting the opening paragraph if you still get this warning you can try play around with whitespace. 

E.g.


```
---
---

{% raw %}
There is...
```

Or 

```
---
---
{% raw %}There
```
Or maybe just start the raw tag after the first paragraph.


```
---
---
First paragraph here

{% raw %}
Second paragraph and rest of tutorial here
...

{% endraw %}
```

Reply:
Starting the raw tag after the first paragraph fixed the warning. But, I noticed that because the whole post is wrapped in the tag, URLs to internal pages are not generated properly.

Reply:
Sorry I should have checked on what would break before making the suggestion 

You'll have to use the raw tag throughout but I would recommend keeping it readable.

e.g. not squeezed in one line 

```
{% raw %}Text
```

And not this either before raw tag actually renders as an empty line

    ```liquid
    {% raw %}
    Text
    ```


But this should be good with raw on the outside.

    {% raw %}
    ```liquid
    Text

Reply:
Done.